sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java

```diff
@@ -92,23 +160,25 @@
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<AttestationResponse> attestSgxEnclave(AttestSgxEnclaveRequest request) {
-        return this.serviceClient.attestSgxEnclaveAsync(request.toGenerated())
-            .map(AttestationResponse::fromGenerated);
+        return withContext(context -> attestSgxEnclaveWithResponse(request, context))
+            .map(Response::getValue);
     }
 
     /**
      * Processes attestation evidence from a VBS enclave, producing an attestation result. The attestation result
      * produced is dependent upon the attestation policy.
      *
      * @param request Attestation request for Trusted Platform Module (TPM) attestation.
+     * @param context - Context for the operation
      * @throws IllegalArgumentException thrown if parameters fail the validation.
      * @throws CloudErrorException thrown if the request is rejected by server.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
      * @return attestation response for Trusted Platform Module (TPM) attestation.
      */
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<String>> attestTpmWithResponse(String request) {
-        return this.serviceClient.attestTpmWithResponseAsync(new com.azure.security.attestation.implementation.models.TpmAttestationRequest().setData(request.getBytes(StandardCharsets.UTF_8)))
+//    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<String>> attestTpmWithResponse(String request, Context context) {
+        Objects.requireNonNull(request);
```

Comment:
We try to avoid client side validation unless the argument in used in the API Route etc, which will result in a poor error experience for the customer.

Reply:
Also curious - what does the response string look like? Can you share an example?

Reply:
To confirm: it's better to have the code crash in the client than validate in the implementation? That's surprising, but if that's the expectation, so be it (the reason I added the requireNonNull is because the compiler complained that this crashes:

```
        return this.attestImpl.attestTpmWithResponseAsync(new com.azure.security.attestation.implementation.models.TpmAttestationRequest().setData(request.getBytes(StandardCharsets.UTF_8)), context)
```

Or are you saying I should only check in the one function that's going to crash and not the others?

Reply:
By implementation I meant the service inner workings. We should definitely add validations for client side crashes but should avoid validation that may result in service returning error responses - just because those are hard to maintain and keep track of - when service invariably changes its behavior.

Reply:
Let's follow this discussion up in email if you don't mind. Not specifically around this validation, but I have other client side validations that will come online very soon (as soon as this pull request is completed :)) that might be relevant.

Reply:
Sounds good.